metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py

```diff
@@ -736,6 +749,9 @@ def add_preparsed_query(
         session_has_temp_tables: bool = True,
         _is_internal: bool = False,
     ) -> None:
+
+        self._tool_meta_extractor.extract_bi_metadata(parsed)
```

Comment:
this is a bit tricky - we only need it here, since observed query calls this? 

imo it's a bit confusing because ObservedQuery also has extra_info, but that isn't copied over by `add()` or `add_observed_query`

Reply:
Right. 

How about removing `extra_info` entirely. I do not believe we are doing anything with `user_via` just yet.

Also I believe we could place the line `self._tool_meta_extractor.extract_bi_metadata(parsed)` in `add()` right before invoking `add_preparsed_query` or `add_observed_query` instead of here and also refractor `PreparsedQuery` / `ObservedQuery` to support base protocol - renaming of `query_text`  to `query`, etc. What do you think? Would it be less confusing then ?

Reply:
Added some refactoring + comment + passing extra_info to PreParsedQuery.